lib/dash/content_protection.js

```diff
@@ -262,6 +262,180 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
+
+    // Convert from bytes to WORD length
+    const wordLength = byteLength / 2;
+    const end = wordLength + head + 2;
+
+    // subarray end is exclusive
+    // + 2 to skip over type and byteLength
+    const recordValue = recordData.subarray(head + 2, end);
+
+    // const recordString = shaka.util.StringUtils.fromUTF8(recordValue);
+    records.push({
+      type: type,
+      value: recordValue,
+    });
+
+    head = end;
+  }
+
+  return records;
+};
+
+
+/**
+* Parses a Uint16Array for PlayReady Objects.  The data
+* should contain a 32-bit integer indicating the length of
+* the PRO in bytes.  Following that, a 16-bit integer for
+* the number of PlayReady Object Records in the PRO.  Lastly,
+* a byte array of the PRO Records themselves.
+*
+* PlayReady Object format: https://goo.gl/W8yAN4
+*
+* @param {!Uint16Array} data
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsPro_ = function(data) {
+  let offset = 0;
+  const view = new DataView(data.buffer);
+  // First 4 bytes is the PRO length
+  const byteLength = view.getUint32(offset, true /* littleEndian */);
+  offset += 2;
+  if (byteLength / 2 !== data.length) {
+    // Malformed PRO
+    return [];
+  }
+
+  // Skip PRO Record count
+  offset++;
+
+  // Rest of the data contains the PRO Records
+  const records = data.subarray(offset);
+  const ContentProtection = shaka.dash.ContentProtection;
+  return ContentProtection.parseMsProRecords_(records);
+};
+
+
+/**
+* PlayReady Header format: https://goo.gl/dBzxNA
+*
+* @param {!Element} xml
+* @return {string}
+* @private
+*/
+shaka.dash.ContentProtection.getLaurl_ = function(xml) {
+  // LA_URL element is optional and no more than one is
+  // allowed inside the DATA element. Only absolute URLs are allowed.
+  // If the LA_URL element exists, it must not be empty.
+  const laurlNode = xml.querySelector('DATA > LA_URL');
+  if (laurlNode) {
+    return laurlNode.textContent;
+  }
+
+  // Not found
+  return '';
+};
+
+
+/**
+* Gets a PlayReady license URL from a content protection element
+* containing a PlayReady Header Object
+*
+* @param {shaka.dash.ContentProtection.Element} element
+* @return {string}
+*/
+shaka.dash.ContentProtection.getPlayReadyLicenseUrl = function(element) {
+  const proNode = shaka.util.XmlUtils.findChild(element.node, 'mspr:pro');
+  if (!proNode) {
+    return '';
+  }
+  const ContentProtection = shaka.dash.ContentProtection;
+  const PLAYREADY_RECORD_TYPES = ContentProtection.PLAYREADY_RECORD_TYPES;
+  const bytes = shaka.util.Uint8ArrayUtils.fromBase64(proNode.textContent);
+  const records = ContentProtection.parseMsPro_(new Uint16Array(bytes.buffer));
+  const record = records.filter((record) => {
+    return record.type === PLAYREADY_RECORD_TYPES.RIGHTS_MANAGEMENT;
+  })[0];
+
+  if (!record) {
+    return '';
+  }
+
+  const parser = new DOMParser();
+  const xmlString = shaka.util.StringUtils.fromUTF8(record.value);
```

Comment:
You should move `shaka.dash.MpdUtils.parseXml` to XmlUtils and use that instead.  That will catch parse errors since `DOMParser` doesn't throw exceptions for invalid XML.